### ISSUE TITLE
[QUESTION] What's the Exclusion Glob patterns syntax?

### ISSUE THREAD

**Issue by RaptorX** [thqby/vscode-autohotkey2-lsp#577]
I have setup some exclusion patterns for a given folder like this:

![20240823@20-03-19](https://github.com/user-attachments/assets/e716337f-cfc3-4236-829b-2cb8e8dfeb4e)

But the files are being read by the extension anyways. I tried a few patterns as you can see in the image.
Am I writing them wrong or the extension is not ignoring the files?

**Comment by thqby** [comment]
This option does not take effect for files opened by the editor or files included by opened files.

**Comment by RaptorX** [comment]
There are not open.

When i open VSCode they get marked red because they were moved to the "old" folder.

I dont want the extension to mark them as problems. And I dont open them (they are copied here by other people in my team), I want the extension to totally ignore them.

But even with the exclusion patterns, they are being read and marked as "having errors" by the extension anyways.

**Comment by thqby** [comment]
**\old\*.*

**Comment by RaptorX** [comment]
> `**\old\*.*`

LOL

Exactly the one that i didnt try... let me check and see if it works on my end :)

**Comment by thqby** [comment]
The open file does not refer to them indirectly? I can't reproduce the issue.

**Comment by RaptorX** [comment]
> The open file does not refer to them indirectly? I can't reproduce the issue.

No they do not.

On the last video I sent, towards the end you can see the whole contents of the "test" file. It doesn't even have includes. Hehe

**Comment by RaptorX** [comment]
> The open file does not refer to them indirectly? I can't reproduce the issue.

Assume
ScriptDir/
    Lib/Lib1.ahk
    Old/
    File1.ahk

File1 includes Lib1.
Duplicate File1 to File2... Same data.
Move File1 to old. Now the include path is invalid (no lib folder in old).

File 2 is your main script now. 

I personally use git for version control but others in my team keep putting files in an "old" folder. 

I can . gitignore and I can ignore them in searches. But the extension keeps making them red and listing all the "invalid path" or "variable not defined" errors even if I try to use the glob ignore patterns.

I just need to see actual errors in my work folder but sometimes it is annoying.

**Comment by thqby** [comment]
![image](https://github.com/user-attachments/assets/a447dea3-c197-48c2-a511-dd00f3f2b814)
Disable other extensions and try again? I still can't reproduce it.

**Comment by RaptorX** [comment]
Update:

The issue only happens when AHK++ from @mark-wiemer is installed but it is not directly related to it.

I marked all the affected files with `#Requires AutoHotkey v2.0`. 

When I have AHK++ disabled all of them get ignored.
When I enable AHK++ they all get marked with errors even if I am ignoring the whole folder.

The following test is with all extensions disabled except for Autohotkey v2 Language Support:

https://github.com/user-attachments/assets/632b8648-c5d4-4653-b13c-c2b81d511ffa

**Comment by thqby** [comment]
https://github.com/mark-wiemer-org/ahkpp/blob/main/src/parser/parser.ts#L40

ahk++ opens all ahk files in the workspace, so the extension is notified of the file opening and starts parsing the script.

**Comment by RaptorX** [comment]
> https://github.com/mark-wiemer-org/ahkpp/blob/main/src/parser/parser.ts#L40
> 
> ahk++ opens all ahk files in the workspace, so the extension is notified of the file opening and starts parsing the script.

Interesting. 

So. I mainly code in v2 and have `ahk` files default to use your extension. Is ahk++ ignoring what the default extension for a given file is? 

I think It shouldn't.

**Comment by mark-wiemer** [comment]
Hi, you'll see I've opened https://github.com/mark-wiemer-org/ahkpp/issues/488. This was an oversight when I added automatic version switching and unique icons for AHK v1 and AHK v2. Because I wanted the icons to be the appropriate color on start, I make the extension scan every file with `.ahk`, `.ahk2`, and other relevant file extensions to check for a `#Requires` directive near the start of the file. This "opens" the file from the perspective of VS Code (though users never see it), and thqby's extension loads every "opened" file regardless of whether it's excluded or not.

I'll be working on a fix this month, but it's a busy next few weeks for me. Sorry for the inconvenience, I'd recommend disabling AHK++ in the short term and watching the issue if this behavior is frustrating.
